src/node/txreconciliation.h

```diff
@@ -76,6 +76,35 @@ class TxReconciliationTracker
     ReconciliationRegisterResult RegisterPeer(NodeId peer_id, bool is_peer_inbound,
                                               uint32_t peer_recon_version, uint64_t remote_salt);
 
+    /**
+     * Step 1. Add new transactions we want to announce to the peer to the local reconciliation set
+     * of the peer, so that those transactions will be reconciled later.
+     * The caller *must* check that the peer is registered for reconciliations.
+     */
+    void AddToSet(NodeId peer_id, const std::vector<uint256>& txs_to_reconcile);
+
+    /**
+     * Before Step 2, we might want to remove a wtxid from the reconciliation set, for example if
+     * the peer just announced the transaction to us.
+     * The caller *must* check that the peer is registered for reconciliations.
+     */
+    void TryRemovingFromSet(NodeId peer_id, const uint256& wtxid_to_remove);
+
+    /**
+     * Step 2. If it's time to request a reconciliation from the peer, this function will return
+     * the details of our local state, which should be communicated to the peer so that they better
+     * know what we need:
+     * - size of our reconciliation set for the peer
+     * - our q-coefficient with the peer, formatted to be transmitted as integer value
+     * If the peer was not previously registered for reconciliations, returns nullopt.
+     */
+    std::optional<std::pair<uint16_t, uint16_t>> MaybeRequestReconciliation(NodeId peer_id);
```

Comment:
```suggestion
    std::optional<std::pair<uint16_t, uint16_t>> MaybeRequestReconciliation(NodeId peer_id, std::chrono::microsecond now);
```

Passing in the time here would be nice for testing. 
* We only have one global mock time currently (globals are always a bit annoying when writing unit/fuzz tests, as you have to reset state after each test)
* The current global mock time is only accurate in intervals of seconds